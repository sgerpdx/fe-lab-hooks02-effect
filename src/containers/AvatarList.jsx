import React, { useState } from 'react';

export default function AvatarList() {
  //const [charList, setCharList] = useState();
  const [loading, setLoading] = useState(true);

  if (loading) return <h2>Loading...</h2>;

  //   return(
  //       <>

  //       </>
  //   )
}
