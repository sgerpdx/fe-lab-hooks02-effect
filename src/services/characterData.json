[
  {
    "_id": "5cf5679a915ecad153ab68d1",
    "allies": ["Riley"],
    "enemies": [],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201",
    "name": "Analay",
    "affiliation": "Unnamed team"
  },
  {
    "_id": "5cf5679a915ecad153ab68d5",
    "allies": ["His "],
    "enemies": ["Korra "],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/6/6f/Arik.png/revision/latest?cb=20140808184002",
    "name": "Arik",
    "affiliation": "Earth Kingdom Air Force"
  },
  {
    "_id": "5cf5679a915ecad153ab68e2",
    "allies": ["Suyin Beifong"],
    "enemies": ["Metal Clan "],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/4/45/Baatar_Jr..png/revision/latest?cb=20150409193510",
    "name": "Baatar Jr.",
    "affiliation": " Beifong family Military of the Earth Empire (formerly) Metal Clan"
  },
  {
    "_id": "5cf5679a915ecad153ab68e9",
    "allies": ["Asami Sato"],
    "enemies": ["Amon"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/d/d4/Bolin.png/revision/latest?cb=20150406232314",
    "name": "Bolin",
    "affiliation": " Fire Ferrets (formerly) Kuvira's army (deserted) Team Avatar Triple Threat Triad (formerly) Republic City Police"
  },
  {
    "_id": "5cf5679a915ecad153ab68fd",
    "allies": ["Ozai"],
    "enemies": ["Earth Kingdom"],
    "name": "Chan (Fire Nation admiral)",
    "affiliation": "Fire Nation Navy"
  },
  {
    "_id": "5cf5679a915ecad153ab6900",
    "allies": ["Aang"],
    "enemies": ["Azula"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/8/81/Chit_Sang.png/revision/latest?cb=20121021000707",
    "name": "Chit Sang"
  },
  {
    "_id": "5cf5679a915ecad153ab68d7",
    "allies": ["Hiroshi Sato", "Korra", ""],
    "enemies": ["Amon"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/b/b6/Asami_Sato.png/revision/latest?cb=20150407110150",
    "name": "Asami Sato",
    "affiliation": " Future Industries Sato family Team Avatar"
  },
  {
    "_id": "5cf5679a915ecad153ab68f0",
    "allies": ["Bumi"],
    "enemies": [],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/b/b6/Bum-Ju.png/revision/latest?cb=20140730103156",
    "name": "Bum-Ju",
    "affiliation": "Spirit World"
  },
  {
    "_id": "5cf5679a915ecad153ab68f5",
    "allies": ["Asami Sato"],
    "enemies": [],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/8/82/Butler.png/revision/latest?cb=20121107115044",
    "name": "Butler",
    "affiliation": "Sato family"
  },
  {
    "_id": "5cf5679a915ecad153ab68ed",
    "allies": [],
    "enemies": ["Zuko"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/9/98/Broadsword_man.png/revision/latest?cb=20130630171553",
    "name": "Broadsword man",
    "affiliation": "Earth Kingdom"
  },
  {
    "_id": "5cf5679a915ecad153ab68cf",
    "allies": ["Red Lotus"],
    "enemies": ["Suyin Beifong"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/0/07/Aiwei_reveals_Korra%27s_lie.png/revision/latest?cb=20140808130551",
    "name": "Aiwei",
    "affiliation": " Metal Clan (formerly) Red Lotus (formerly)"
  },
  {
    "_id": "5cf5679a915ecad153ab68d2",
    "allies": ["The butcher"],
    "enemies": ["Momo"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/5/58/Animal_control_officer.png/revision/latest?cb=20140504160655",
    "name": "Animal control officer"
  },
  {
    "_id": "5cf5679a915ecad153ab6904",
    "allies": ["Tu"],
    "enemies": [],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/5/51/Chow.png/revision/latest?cb=20140730173827",
    "name": "Chow"
  },
  {
    "_id": "5cf5679a915ecad153ab6906",
    "allies": ["Ty Lee"],
    "enemies": ["Appa"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/a/a5/Circus_master.png/revision/latest?cb=20130706153819",
    "name": "Circus master",
    "affiliation": "Fire Nation circus"
  },
  {
    "_id": "5cf5679a915ecad153ab68ee",
    "allies": ["Ozai"],
    "enemies": ["Zuko"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/7/76/Bujing.png/revision/latest?cb=20130714152817",
    "name": "Bujing",
    "affiliation": "Fire Nation military"
  },
  {
    "_id": "5cf5679a915ecad153ab68f6",
    "allies": ["Raiko", ""],
    "enemies": ["Varrick"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/e/ed/Buttercup_Raiko.png/revision/latest?cb=20150409193510",
    "name": "Buttercup Raiko",
    "affiliation": "United Republic Government"
  },
  {
    "_id": "5cf5679a915ecad153ab68f7",
    "allies": ["Cabbage Corp"],
    "enemies": ["The "],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/2/2f/Cabbage_merchant.png/revision/latest?cb=20140112200908",
    "name": "Cabbage merchant",
    "affiliation": " Cabbage Corp Earth Kingdom"
  },
  {
    "_id": "5cf5679a915ecad153ab68f8",
    "allies": ["Aunt "],
    "enemies": ["Fire Nation"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/b/b6/Calm_man.png/revision/latest?cb=20130814093330",
    "name": "Calm man",
    "affiliation": "Earth Kingdom"
  },
  {
    "_id": "5cf5679a915ecad153ab68ff",
    "allies": ["His military"],
    "enemies": ["Avatar"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/6/6f/Chin_the_Great.png/revision/latest?cb=20140215111604",
    "name": "Chin"
  },
  {
    "_id": "5cf5679a915ecad153ab6901",
    "allies": ["Sokka"],
    "enemies": ["The warden"],
    "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/6/6e/Chit_Sang%27s_friend.png/revision/latest?cb=20121120131829",
    "name": "Chit Sang's friend"
  }
]
