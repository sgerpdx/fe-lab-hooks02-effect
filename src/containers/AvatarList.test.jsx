import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import AvatarList from './AvatarList';
import userEvent from '@testing-library/user-event';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import characterData from '../services/characterData.json';

const avatarServer = setupServer(
  rest.get(
    'https://last-airbender-api.herokuapp.com/api/v1/characters',
    (req, res, ctx) => {
      return res(ctx.json(characterData));
    }
  )
);

describe('Functional Container Component', () => {
  beforeAll(() => avatarServer.listen());
  afterAll(() => avatarServer.close());

  it('displays a list of Avatar characters via AvatarList', async () => {
    render(<AvatarList />);

    // Testing for page loading and basic elements:
    const loading = screen.getByText('Loading...');
    expect(loading).toMatchSnapshot();
    const list = screen.findByRole('list');
    expect(list).toMatchSnapshot();
  });

  // Testing for correct data and shape:
  return waitFor(() => {
    expect(characterData[0]).toEqual({
      _id: '5cf5679a915ecad153ab68d1',
      allies: ['Riley'],
      enemies: [],
      photoUrl:
        'https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201',
      name: 'Analay',
      affiliation: 'Unnamed team',
    });
    screen.getByText('⚡ click on a picture for detail ⚡');
  });
});
