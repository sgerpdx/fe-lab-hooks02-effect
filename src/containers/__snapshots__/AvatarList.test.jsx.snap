// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functional Container Component displays a list of Avatar characters via AvatarList 1`] = `
<h2>
  Loading...
</h2>
`;

exports[`Functional Container Component displays a list of Avatar characters via AvatarList 2`] = `Promise {}`;
